static/src/javascripts/projects/common/modules/user-prefs.spec.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import userPrefs from 'projects/common/modules/user-prefs';
+import { local } from 'lib/storage';
+
+class LocalStorageMock {
+    store: Object;
+
+    constructor() {
+        this.store = {};
+    }
+
+    clear() {
+        this.store = {};
+    }
+
+    getItem(key) {
+        return this.store[key];
+    }
+
+    setItem(key, value) {
+        this.store[key] = value.toString();
+    }
+
+    removeItem(key) {
+        this.store[key] = null;
+    }
+}
+
+const localStorage = new LocalStorageMock();
+
+describe('userPrefs - Client-side preferences', () => {
+    beforeAll(() => {
+        local.storage = localStorage;
```

Comment:
Yes, I find it frustrating that Jest conflates stubs, mocks and spies into one thing it calls `mocks`. It probably [annoys Martin Fowler](https://martinfowler.com/articles/mocksArentStubs.html) more though.